dm/worker/upgrade.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"encoding/binary"
+	"os"
+	"strconv"
+
+	"github.com/pingcap/errors"
+	"github.com/syndtr/goleveldb/leveldb"
+
+	"github.com/pingcap/dm/pkg/log"
+)
+
+const (
+	// The current internal version of DM-worker used when upgrading from an older version, and it's different from the release version.
+	// NOTE: +1 when an incompatible problem is introduced.
+	currentWorkerVersion uint64 = 1
+	// The default previous internal version of DM-worker if no valid internal version exists in DB before the upgrade.
+	defaultPreviousWorkerVersion uint64 = 0
```

Comment:
In fact, our version may not be linear, more like a forked tree, now you should be able to understand why you don't want to do a version upgrade.

Reply:
added a _release version_ in https://github.com/pingcap/dm/pull/187/commits/11d9b85209462289794102b09ac643411d990951.